clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2265,27 +2560,20 @@ class SyclOptReportCreator : public SyclKernelFieldHandler {
   SourceLocation KernelInvocationLoc;
 
   void addParam(const FieldDecl *KernelArg, QualType KernelArgType,
-                StringRef KernelArgDescription) {
+                StringRef KernelArgDescription,
+                bool IsCompilerGeneratedType = false) {
     StringRef NameToEmitInDescription = KernelArg->getName();
     const RecordDecl *KernelArgParent = KernelArg->getParent();
     if (KernelArgParent && KernelArgDescription == "decomposed struct/class")
       NameToEmitInDescription = KernelArgParent->getName();
 
-    bool isWrappedField = KernelArgDescription == "WrappedPointer" ||
```

Comment:
But I guess we still have wrapped pointers in case the struct containing pointers was decomposed, right?

Reply:
Yes. But the opt-report output for wrapped fields is "Compiler generated" because we don't want implementation details in report. So we can just pass `IsCompilerGeneratedType` for wrapped fields as well.